content/sensu-core/2.0/reference/tokens.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Tokens"
+description: "The tokens reference guide."
+weight: 1
+version: "2.0"
+product: "Sensu Core"
+menu: 
+  sensu-core-2.0:
+    parent: reference
+---
+
+## How do tokens work?
+When a check is scheduled to be executed by an agent, it first goes through a token substitution step. Any tokens matching attribute values in the check are applied, and then the check is executed. Invalid templates or unmatched tokens will return an error, which is logged and sent to the Sensu backend message transport. Checks with token matching errors will not be executed.
+
+## New and improved tokens
+Sensu 2.0 uses the [Go template][1] package to implement token substitution. Instead of using triple colons `:::` as in [1.x token substitution][2], 2.0 token substitution uses double curly braces `{{` around tokens to be substituted.
+
+## Sensu tokens specification
+
+### Token substitution syntax
+
+Tokens are invoked by wrapping references to entity or custom attributes with double curly braces, such as `{{ .ID }}` to substitute an entity's ID value. Nested Sensu [entity attributes][3] can be accessed via dot notation (e.g. `system.arch`).
+
+- `{{ system.network.interfaces.addresses }}` would be replaced with the [entity `system.network.interfaces.address` attribute][26]
```

Comment:
s/system.network.interfaces.addresses/System.Network.Interfaces.Addresses/

Also, since `Addresses` is a slice of strings, I'm not totally what would be the result though.

Reply:
Yeah, I was not quite sure what to put for that either - was attempting to translate the 1.x examples to 2.x format for docs consistency. I'll play around with this a bit and see if I can come up with a better example.

Reply:
I completely forgot to mention that we already had some docs for the tokens here: https://github.com/sensu/sensu-alpha-documentation/blob/master/06-checks-and-assets.md#token-substitution

Really sorry about that :(

Reply:
No worries, I used those as a reference!